internal/config/load.go

```diff
@@ -16,15 +17,38 @@ import (
 const REMOTE_CONFIG_CHECK_TIME_KEY = "REMOTE_CONFIG_CHECK_TIME"
 const REMOTE_CONFIG_CHECK_CYCLE_KEY = "REMOTE_CONFIG_CHECK_CYCLE"
 
+// store some metadata about the configuration settings
+//
+type ConfigMetadata struct {
+	File   string
+	Reason string
+}
+
+var (
+	configMetadata = ConfigMetadata{}
+)
+
 func LoadConfig(appCtx context.LauncherContext) {
 	// NOTE: we don't put default value for the DEBUG_FLAGS configuration, it will not show in a newly created config file
 	// Please keep it as a hidden config, better not to let developer directly see this option
 	setDefaultConfig()
-
+	wd, _ := os.Getwd()
 	cfgFile := os.Getenv(appCtx.ConfigurationFileEnvVar())
+	localCftFileName := fmt.Sprintf("%s.json", appCtx.AppName())
 	if cfgFile != "" {
+		configMetadata.Reason = fmt.Sprintf("from environment variable: %s", appCtx.ConfigurationFileEnvVar())
+		configMetadata.File = cfgFile
+
 		viper.SetConfigFile(cfgFile)
+	} else if localCfgPath, found := findLocalConfig(wd, localCftFileName); found {
+		configMetadata.Reason = "found config file from working dir or its parents"
+		configMetadata.File = localCfgPath
+
+		viper.SetConfigFile(localCfgPath)
 	} else {
+		configMetadata.Reason = fmt.Sprintf("use default config file from app home %s", AppDir())
```

Comment:
It's 3 times AppDir() invocation. AppDir consists of several operation(reading env data, tracing), which seem to be redundant just to put directory in these 3 location below

Reply:
Fixed